src/Parser/Parts/PartList.cs

```diff
@@ -10,6 +10,23 @@ public PartList()
         {
         }
 
+ 
+        
+        /// <summary>
+        /// Number of <see cref="TextPart"/> with type <see cref="HoleType.Numeric"/>
+        /// </summary>
+        public int NumericPartCount { get; set; }
```

Comment:
What do we need those counts for? I think we only need to know if there is at least one non-numeric part, right? That could be just a bool flag set when parsing.